testpar/t_file.c

```diff
@@ -106,7 +106,9 @@ test_split_comm_access(void)
 
         /* delete the test file */
         if (sub_mpi_rank == 0) {
-            mrc = MPI_File_delete((char *)filename, info);
+            char fname[NAME_MAX];
+            strncpy(fname, filename, strlen(filename));
+            mrc = MPI_File_delete(fname, info);
```

Comment:
Looks like you can delete the cast to `char *` ?

Reply:
Agreed. MPI_File_delete takes a const char *, so no need to do a strncpy here, just drop the char * cast.